### PR TITLE
Assure basedir exists before creating resultdir

### PR DESCRIPTION
**Pull request by praiskup**
Because resultdir is, by default, created below basedir.  This has been breaking use-cases started with commands.init(do_log=True).

The bug was caused by f7b0c9bf18c46046be922adc740d1c6616ae0cc6, where we started pre-creating resuldir sooner than before so we could start the logging early.

Fixes: #1183
Complements: f7b0c9bf18c46046be922adc740d1c6616ae0cc6

### CODE REVIEW

**Review by Conan-Kudo — APPROVED** [review]

File: mock/py/mockbuild/buildroot.py [hunk 1]
```diff
@@ -40,6 +40,18 @@ def wrapper(self, *args, **kwargs):
...
 
 
+def call_just_once(f):
+    """
+    Make sure that an object method is called just once.
+    """
+    def wrapper(self, *args, **kwargs):
+        if hasattr(self, "called_method_retval"):
+            return self.called_method_retval
+        self.called_method_retval = f(self, *args, **kwargs)
```
Comment (QuLogic) [inline mock/py/mockbuild/buildroot.py:50]: This is a bit odd; you're saving the return value on `self`, which means that there is only ever one return value saved regardless of how many functions are decorated. It's also given a public naming convention even though it's not public.

File: mock/py/mockbuild/buildroot.py [hunk 2]
```diff
@@ -40,6 +40,18 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
+def call_just_once(f):
+    """
+    Make sure that an object method is called just once.
+    """
+    def wrapper(self, *args, **kwargs):
```
Comment (QuLogic) [inline mock/py/mockbuild/buildroot.py:47]: Best practice would be to decorate this with `functools.wraps`.

File: mock/py/mockbuild/buildroot.py (hunk 1 above)
Comment (QuLogic) [reply mock/py/mockbuild/buildroot.py:50]: Also odd is that this passes through `*args, **kwargs`, but that is not used to differentiate return values.

Comment (praiskup) [reply mock/py/mockbuild/buildroot.py]: Good point. This would be very tricky in the future, thank you for the review! TIL

**Pull request merged**